gammapy/analysis/core.py

```diff
@@ -331,38 +356,44 @@ def _map_making(self):
 
         stacked = MapDataset.create(geom=geom, name="stacked", **geom_irf)
 
-        if datasets_settings.stack:
-            for obs in self.observations:
-                log.info(f"Processing observation {obs.obs_id}")
-                cutout = stacked.cutout(obs.pointing_radec, width=2 * offset_max)
-                dataset = maker.run(cutout, obs)
-                dataset = maker_safe_mask.run(dataset, obs)
-                if bkg_maker is not None:
-                    dataset = bkg_maker.run(dataset)
-
-                if bkg_method == "ring":
-                    dataset = dataset.to_map_dataset()
-
-                log.debug(dataset)
-                stacked.stack(dataset)
-            datasets = [stacked]
-        else:
-            datasets = []
-
-            for obs in self.observations:
-                log.info(f"Processing observation {obs.obs_id}")
-                cutout = stacked.cutout(obs.pointing_radec, width=2 * offset_max)
-                dataset = maker.run(cutout, obs)
-                dataset = maker_safe_mask.run(dataset, obs)
-                if bkg_maker is not None:
-                    dataset = bkg_maker.run(dataset)
-                log.debug(dataset)
-                datasets.append(dataset)
-
+        with pbar(total=len(self.observations), show_pbar=show_pbar, desc="Datasets") as pb:
+            if datasets_settings.stack:
+                for obs in self.observations:
```

Comment:
I think we can keep the log.info() if show_pbar is False

```
if not show_pbar:
    log.info(f"Processing observation {obs.obs_id}")
```

Reply:
I downgraded the log priority to `debug` instead. Do you think it's ok? This way, the user can still see the processed observation ID (even breaking the progress bar) if really needed.

Possibly, as suggested by @registerrier during yesterday's call, we might even show the progress bar only when the log level is set to a high verbosity. This would allow to remove the boolean `show_pbar`. I like this idea and I think it is way to go, but it requires a good cleanup of the logging that we do in gammapy. It is work for the future, I guess

Reply:
>  downgraded the log priority to debug instead.

I do not understand what you mean with this, sorry..
when using the HLI the log level is [set to INFO by default](https://github.com/gammapy/gammapy/blob/master/gammapy/analysis/config.py#L212). So, all logs with level info and higher will be displayed. 

I don't think automatically showing the progress bar in debug level would be ok, on the contrary I would display progress bar if log is set to warnings, so only high severity logs will break the display.

Reply:
> downgraded the log priority to debug instead.

ok, now that I can see your modifs, I know what you mean.
fine !